warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/evaluation/EdgePreconditionArithmetic.java

```diff
@@ -0,0 +1,295 @@
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            Iterator iter = ((List) left).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.lessThanOrEqual(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+            
+        } else if (right instanceof Set) {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            
+            Iterator iter = ((List) right).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newRight = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.lessThanOrEqual(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+        }
+        
+        else {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            if (super.lessThanOrEqual(newLeft, newRight)) {
+                addMatchingGroup(newLeft);
+                addMatchingGroup(newRight);
+                matches = true;
+                
+            }
+        }
+        
+        return matches;
+    }
+    
+    @Override
+    public boolean greaterThan(final Object left, final Object right) {
+        boolean matches = false;
+        
+        if (left instanceof List && !(right instanceof List)) {
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            Iterator iter = ((List) left).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.greaterThan(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+            
+        } else if (right instanceof Set) {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            
+            Iterator iter = ((List) right).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newRight = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.greaterThan(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+        }
+        
+        else {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
```

Comment:
Why are we assuming a long value here?

Reply:
We currently don't have any preconditions requiring gt, lt, ge, le operators, but you'd suggested I add the methods here anyway.  We also currently don't use the normalizers when comparing values for preconditions, though there is another ticket to do so.  Since non-numeric comparisons here don't make sense, we don't have any need or requests for them yet, and this is supposed to be a quick bug fix, I opted to supply minimal functionality here.  I will make it obvious that non-numeric gt, lt, le, ge comparisons are not supported for now and address them under the other ticket later.